### PR TITLE
Fix iOS 15 default blue color and add background color to close icon [DDFLSBP-414]

### PR DESCRIPTION
**Pull request by clausbruun**
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-414

#### Description

* Adding a background color to the close icon in modals to prevent interference with the content in the modal.
* Setting a specific color for texts that would otherwise adopt the default iOS 15 blue color.
* Transferring changes made to the user icon in the header from dpl-react to the dpl-design-system.

#### Screenshot of the result

_Close icon background color:_

https://github.com/danskernesdigitalebibliotek/dpl-design-system/assets/105956/9d584289-2c3f-4073-be34-c70e0d0352e0

_No blue texts:_

image

image

#### Additional comments or questions

*

### CODE REVIEW

File: src/stories/Library/Lists/list-reservation/list-reservation.scss [hunk 1]
```diff
@@ -18,6 +18,14 @@ $list-reservation-space-desktop: 24px;
   padding: $list-reservation-space-mobile;
   cursor: pointer;
 
+  h3 {
+    color: $color__text-primary-black;
```
Comment (Adamik10) [inline src/stories/Library/Lists/list-reservation/list-reservation.scss:22]: Very nice that you're using the variables 👏 💯

File: src/stories/Blocks/header/Header.tsx [hunk 2]
```diff
@@ -109,21 +109,19 @@ export const Header = (props: HeaderProps) => {
...
-                {signedIn && haveNotification && (
-                  <div className="header__notification bg-signal-alert" />
-                )}
-                <ProfileIcon />
-                {signedIn && (
-                  <span className="text-small-caption">{username}</span>
-                )}
-              </a>
-            </div>
+            <button
```
Comment (Adamik10) [inline src/stories/Blocks/header/Header.tsx:112]: 👏 Great that you're bringing the actual implementation back to the design system 👏

**Review by Adamik10 — APPROVED** [review]
Looks perf! You just need to check out your Chromatic UI tests. We need to talk to Kasper G if you don't have the rights to approve them (note that you have to sign in with. your GH user in there)

File: src/stories/Library/Lists/list-reservation/list-reservation.scss (hunk 1 above)
Comment (clausbruun) [reply src/stories/Library/Lists/list-reservation/list-reservation.scss:22]: Thank you @Adamik10 🙏🏻

**Comment by clausbruun** [comment]
> The bottom border doesn't align with the rest of the borders now. Do you know why that could be? I would recommend that to be solved before we move this forward 🙈

I'm sorry, I should have caught that myself. 🫠

**Comment by clausbruun** [comment]
The border property was [inherited](https://github.com/danskernesdigitalebibliotek/dpl-design-system/blob/64108bc1ebfea865e123604274e100a03902ce91/src/styles/scss/reset.scss#L72) from [this element](https://github.com/danskernesdigitalebibliotek/dpl-design-system/blob/64108bc1ebfea865e123604274e100a03902ce91/src/stories/Blocks/header/header.scss#L45), so I have added a fix for that.

**Pull request merged**